great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py

```diff
@@ -37,30 +77,141 @@ def __init__(
         if not isinstance(meta, dict):
             raise ge_exceptions.ProfilerExecutionError(
                 message=f"""Argument "{meta}" in "{self.__class__.__name__}" must be of type "dictionary" \
-(value of type "{str(type())}" was encountered).
+(value of type "{str(type(meta))}" was encountered).
 """
             )
 
+        if condition and (not isinstance(condition, str)):
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"""Argument "{condition}" in "{self.__class__.__name__}" must be of type "string" \
+(value of type "{str(type(condition))}" was encountered).
+"""
+            )
+
+        self._condition = condition
         self._meta = meta
 
     @property
     def expectation_type(self) -> str:
         return self._expectation_type
 
     @property
-    def kwargs(self) -> dict:
-        return self._kwargs
+    def condition(self) -> str:
+        return self._condition
 
     @property
     def meta(self) -> dict:
         return self._meta
 
+    @property
+    def kwargs(self) -> dict:
+        return self._kwargs
+
+    def _parse_condition(self) -> ParseResults:
+        """
+        Using the grammer defined by expr, provides the condition to the parser.
+
+        Applicability: To be used as part of configuration (e.g., YAML-based files or text strings).
+        Extendability: Readily extensible to include "slice" and other standard accessors (as long as no dynamic elements).
+        """
+
+        try:
+            return expr.parseString(self._condition)
+        except ParseException:
+            raise ExpectationConfigurationConditionParserError(
+                f'Unable to parse Expectation Configuration Condition: "{self._condition}".'
+            )
+
+    def _substitute_parameters_and_variables(
```

Comment:
@NathanFarmer A docstring for `_substitute_parameters_and_variables()` might be helpful.  What is its purpose?  What does it do?  There is already one below, but I am curious about "why is it needed?", "what is the meaning of term-list", etc.  Thanks.

Reply:
@alexsherstinsky I have added more color to the docstrings. Please let me know if about any remaining questions you might have.